src/backends/backend.js

```diff
@@ -250,11 +269,7 @@ class Backend {
       }
       const slug = slugFormatter(collection.get("slug"), entryDraft.getIn(["entry", "data"]), config.get("slug"));
       const path = selectEntryPath(collection, slug);
-      entryObj = {
-        path,
-        slug,
-        raw: this.entryToRaw(collection, entryDraft.get("entry")),
-      };
+      entryObj = await this.checkOverwrite(collection, slug, path, entryDraft);
```

Comment:
Instead of implicitly constructing and returning the entry object from `checkOverwrite`, we should construct the entry object here either before or after the check.

Reply:
Working on an update commit.